Ecs/src/SparseArray.hpp

```diff
@@ -0,0 +1,308 @@
+        };
+
+        /**
+         * @brief Returns a read-only (constant) iterator that points one past
+         *  the last element in the sparseArray. Iteration is done in
+         *  ordinary element order.
+         *
+         * @return const_iterator.
+         */
+        const_iterator cend() const
+        {
+            return _data.cend();
+        };
+
+        /**
+         * @brief Returns the number of elements in the sparseArray.
+         *
+         * @return size_type of the sparseArray.
+         */
+        size_type size() const
+        {
+            return _data.size();
+        };
+
+        /**
+         * @brief Inserts given value into sparseArray at the specified position.
+         *
+         * @param pos Index to insert element.
+         * @param component Component to insert.
+         * @return reference_type of the new component.
+         */
+        reference_type insert_at(size_type pos, Component const &component)
+        {
+            if (pos >= _data.size()) {
+                _data.resize(pos + 1);
+            }
+            _data[pos] = component;
+            return _data[pos];
+        }
+
+        /**
+         * @brief Inserts given value into sparseArray at the specified position.
+         *
+         * @param pos Index to insert element.
+         * @param component Component to insert.
+         * @return reference_type of the new component.
+         */
+        reference_type insert_at(size_type pos, Component &&component)
+        {
+            if (pos >= _data.size()) {
+                _data.resize(pos + 1);
+            }
+            _data[pos] = component;
+            return _data[pos];
+        };
+
+        /**
+         * @brief Inserts an object in sparseArray before specified iterator.
+         *
+         * @tparam Params for component to create and insert.
+         * @param pos Index to insert element.
+         * @param params Component to insert.
+         * @return reference_type of the new component.
+         */
+        template <class... Params>
+        reference_type emplace_at(size_type pos, Params&&... params)
+        {
+            if (pos >= _data.size()) {
+                _data.resize(pos + 1);
+            }
+            _data[pos] = Component(std::forward<Params>(params)...);
+            return _data[pos];
+        }
+
+        /**
+         * @brief Erase a row in the sparseArray.
+         *
+         * @param pos Index of the element to erase.
+         */
+        void erase(size_type pos)
+        {
+            _data.erase(_data.begin() + pos);
+        };
+
+        /**
+         * @brief Get the index of an object.
+         *
+         * @param value of the object.
+         * @return size_type Index of the object.
+         */
+        size_type get_index(value_type const &value) const
+        {
+            for (size_type i = 0; i < _data.size(); ++i) {
+                if (_data[i] == value)
+                    return i;
```

Comment:
Since in other methods you put brackets for one line execution if, I think, as the "this" specifier, we should specify in the contributing if we are keeping the brackets or not for single line if conditions